docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,226 @@
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+### Managed cluster infrastructure
+
+Cluster infratructure which lifecycle is managed by a provider infraCluster CR.
+E.g in AWS:
+- Network
+    - VPC
+    - Subnets
+    - Internet gateways
+    - Nat gateways
+    - Route tables
+- Security groups
+- Load balancers
+
+### Externally managed cluster infrastructure
+Cluster infratructure which lifecycle is not managed by CAPI but rather by an external entity.
+
+## Summary
+
+This proposal introduces first class support for "externally managed" infrastructure for CAPI providers.
+This consolidates the boundaries between CAPI managed (existing pattern) and externally managed ("Bring Your Own") cluster infrastructure.
+
+## Motivation
+
+Currently, Cluster API infrastructure providers support an opinionated happy path to create and manage cluster infrastructure lifecycle.
+The fundamental use case we want to support is out of tree controllers or tools that can manage this cluster infrastructure.
+
+For example, this could allow users to create clusters using tooling such as Terraform or Kops and allow them to add CAPI Machine infrastructure as a day 2 operation.
+
+This will also ease adoption of CAPI in heterogeneous real world environments with restricted privileges and where the provider infrastructure for the cluster needs to be managed out of band.
+
+### Goals
+
+- Introduce support for "externally managed" infrastructure consistently across CAPI providers.
+- The machine controller must be able to operate and manage machines when the infastructure is "externally managed".
+- Reuse existing InfraCluster CRDs in "externally managed" clusters to minimise differences between the two topologies.
+
+### Non-Goals/Future Work
+
+- Modify existing managed behaviour.
+- Automatically mark InfraCluster resources as ready (this will be up to the external management component initially).
+- Support externally managed machines.
+
+## Proposal
+
+A new contract will be defined around an annotation `cluster.x-k8s.io/managed-by: "<name-of-system>"`.
+This annotation will identify a system which is providing external management of the infrastructure.
+
+It is expected that all providers will adhere to the contract being defined by this proposal.
+
+When this annotation is present on an InfraCluster resource, the InfraCluster controller is expected to ignore the resource and not perform any reconciliation.
+Importantly, it will not modify the resource or its status in any way.
```

Comment:
Should we mention that Cluster API is providing a predicate to filter events out?

Reply:
This is added just below this, can we resolve?